src/lib/Form/TrashLocationOptionProvider/HasChildren.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Form\TrashLocationOptionProvider;
+
+use eZ\Publish\API\Repository\LocationService;
+use eZ\Publish\API\Repository\Values\Content\Location;
+use EzSystems\EzPlatformAdminUi\Specification\Location\HasChildren as HasChildrenSpec;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+final class HasChildren implements TrashLocationOptionProvider
+{
+    /** @var \eZ\Publish\API\Repository\LocationService */
+    private $locationService;
+
+    /** @var \Symfony\Component\Translation\TranslatorInterface */
+    private $translator;
+
+    public function __construct(LocationService $locationService, TranslatorInterface $translator)
+    {
+        $this->locationService = $locationService;
+        $this->translator = $translator;
+    }
+
+    public function supports(Location $location): bool
+    {
+        return (new HasChildrenSpec($this->locationService))->isSatisfiedBy($location);
+    }
+
+    public function addOptions(FormInterface $form, Location $location): void
+    {
+        $locationChildren = $this->locationService->loadLocationChildren($location);
```

Comment:
```suggestion
        $locationChildren = $this->locationService->loadLocationChildren($location, 0, 0);
```

A) There is no need to load 25 children to get `totalCount`. 

B) `loadLocationChildren` result is filtered by permissions so if the user doesn't have permissions to children, the warning will not be displayed at all.

Reply:
in case of B, user is not allowed to trash that location so I don't think that is a problem at all, at least at this moment